app/models/katello/root_repository.rb

```diff
@@ -129,6 +130,12 @@ def ensure_no_download_policy
       end
     end
 
+    def ensure_no_checksum_on_demand
+      if checksum_type.present? && [::Runcible::Models::YumImporter::DOWNLOAD_ON_DEMAND].include?(download_policy)
```

Comment:
Should this just be `::Runcible::Models::YumImporter::DOWNLOAD_ON_DEMAND == download_policy` ?